### ISSUE TITLE
Is the self.encoder learning even after feats.detach()?

### ISSUE THREAD

**Issue by fariasfc** [vturrisi/solo-learn#135]
Calling `feats.detach()` will prevent the gradient from flowing during backpropagation to previous computational nodes, right? Will not it prevent the encoder (ResNet) to learn?

https://github.com/vturrisi/solo-learn/blob/9f1b1315207a2cea66167358565254e60a03b0b6/solo/methods/base.py#L314

**Comment by vturrisi** [comment]
Hey,
`.detach()` is not done in-place, so only for `self.classifier()` the features are detached.
